rust/lance/src/index/cache.rs

```diff
@@ -39,6 +39,12 @@ impl IndexCache {
         self.vector_cache.entry_count() as usize
     }
 
+    pub(crate) fn get_size(&self) -> usize {
+        self.scalar_cache.sync();
+        self.vector_cache.sync();
+        self.scalar_cache.entry_count() as usize + self.vector_cache.entry_count() as usize
```

Comment:
@wjones127 do you feel just summing this is ok?

Reply:
TBH, I’m not sure I care about the entry count. As a user, I would much rather set the limit in terms of bytes and get the total bytes consumed in the statistics. So this is fine for now but long term I think we ought to consider switching to evicting based on in-memory size.

Reply:
IIUC we can use `weighted_size` and with [weigher](https://docs.rs/moka/latest/moka/future/struct.CacheBuilder.html#method.weigher) to do that. (I hope moka is better at cache invaliation than naming).
I can give that a quick try.

Reply:
Actually let's make that a separate issue to not stall this one.

Reply:
> I hope moka is better at cache invaliation than naming
🤣 
> Actually let's make that a separate issue to not stall this one.

Sounds good!

Reply:
Opened #1613

Reply:
I think summing is fine.  We might just merge these into one field someday.

However, I think there is one potential problem.  The user might set the index cache size to X and then, if they have both scalar and vector indices, they might see an entry count of `2 * X`.  Still, the best long term solution is probably to do bytes so let's stick with summing for now.